packages/gateway/src/features/user/root-resolvers.ts

```diff
@@ -267,8 +271,24 @@ export const resolvers: GQLResolver = {
       })
 
       if (res.status === 403) {
+        const errorResponse = await res.json()
+        const duplicateDataErrorMap = {
+          emailForNotification: {
+            field: 'emailForNotification',
+            conflictingValue: userPayload.emailForNotification
+          },
+          mobile: {
+            field: 'mobile',
+            conflictingValue: userPayload.mobile
+          }
+        }
+
         return await Promise.reject(
-          new Error(`DUPLICATE_MOBILE-${userPayload.mobile}`)
+          new Error(
+            JSON.stringify(
+              duplicateDataErrorMap[errorResponse['errorThrowingProperty']]
+            )
+          )
```

Comment:
https://www.apollographql.com/docs/apollo-server/data/errors#including-custom-error-details Could this be helpful?

Reply:
> it seems a bit odd that we need to stringify the JSON and then parse on client. I think we should look into what is the recommended way to pass errors like this to frontend with Apollo. I'm not sure if it's possible with the GraphQL resolvers, but something like this would feel more conventional:
> 
> ```
> return res.status(403).json(duplicateDataErrorMap[errorResponse['errorThrowingProperty']])
> ```
> 
> @Zangetsu101

yes, my intention was to do something similar , didn't find out a way to do this. But the link you shared , that is helpful. we can use this approach.

Reply:
I found some usages of this CustomError named 'UserInputError' provided by 'apollo-server-hapi' , being used in some places and in the user root-resolver as well, which lets us pass an extension

https://github.com/opencrvs/opencrvs-core/blob/5e41f854aab3090abdbc427f1a2de1f0d1d2b311/packages/gateway/src/features/user/root-resolvers.ts#L259 

I have used it for our purpose.